File: 2017/solutions/b_nine_ks.py
def dsum(s):
    res = 0
    for d in s:
        res += ord(d) - ord('0')
    return res

fin = open("nine.in")
fout = open("nine.out", "w")
a = fin.readline().strip()
b = fin.readline().strip()
if dsum(a) * dsum(b) % 9 == 0:
    fout.write("Yes\n")
else:
    fout.write("No\n")

File: 2017/solutions/b_nine_ks.pas
{$H+}
var a, b: string;

function dsum(s : string) : longint;
var i: longint;
begin
    dsum := 0;
    for i := 1 to length(s) do
        dsum := dsum + ord(s[i]) - ord('0');
end;

begin
    assign(input, 'nine.in');
    assign(output, 'nine.out');
    reset(input);
    rewrite(output);

    readln(a);
    readln(b);

    if Int64(dsum(a)) * dsum(b) mod 9 = 0 then
        writeln('YES')
    else
        writeln('NO');

    close(input);
    close(output);
end.
